### PR TITLE
Feature/id 27/monthly summary

### PR DESCRIPTION
**Pull request by ronaldo-mendoza-jalafund**
## Description
The system should provide an endpoint that accepts query parameters for start and end months, returning a list of monthly summaries within that range. Each summary should include the month and year, total expenses, total income, and remaining budget.

## US related
- [US | Monthly Summary](https://trello.com/c/8jCU8IPZ/42-id-27-monthly-summary)

### CODE REVIEW

File: src/Services/UserService.cs [hunk 1]
```diff
@@ -36,4 +42,56 @@ public async Task<User> UpdateUserProfileAsync(Guid userId, UpdateUserDTO user)
         
         return updatedUser;
     }
+
+    public async Task<IEnumerable<MonthlySummaryDTO>> GetMonthlySummaryAsync(Guid userId, DateTime startDate, DateTime endDate)
+    {
+        if (userId == Guid.Empty)
```
Comment (ignacio-villarroel-jalafund) [inline src/Services/UserService.cs:48]: Apply some desing pattern to your code, is not a good practice use to many if statements in a row

File: src/Services/UserService.cs [hunk 2]
```diff
@@ -36,4 +42,56 @@ public async Task<User> UpdateUserProfileAsync(Guid userId, UpdateUserDTO user)
...
+        {
+            throw new ArgumentException("El ID de usuario no puede ser vacío", nameof(userId));
```
Comment (ignacio-villarroel-jalafund) [inline src/Services/UserService.cs]: Change your code, we are creating a project entirely in english

File: src/Services/UserService.cs [hunk 3]
```diff
@@ -36,4 +42,56 @@ public async Task<User> UpdateUserProfileAsync(Guid userId, UpdateUserDTO user)
...
+        }
+
+        if (startDate.Year >= endDate.Year)
+        {
+            throw new ArgumentException("El año de inicio debe ser menor al año de fin", nameof(startDate));
```
Comment (ignacio-villarroel-jalafund) [inline src/Services/UserService.cs]: Change this message too

File: src/Services/UserService.cs [hunk 4]
```diff
@@ -36,4 +42,56 @@ public async Task<User> UpdateUserProfileAsync(Guid userId, UpdateUserDTO user)
...
+        }
+
+        if (startDate.Month >= endDate.Month)
+        {
+            throw new ArgumentException("El mes de inicio debe ser menor al mes de fin", nameof(startDate));
```
Comment (ignacio-villarroel-jalafund) [inline src/Services/UserService.cs]: Also with this, keep the code in english

File: src/Services/UserService.cs [hunk 5]
```diff
@@ -36,4 +42,56 @@ public async Task<User> UpdateUserProfileAsync(Guid userId, UpdateUserDTO user)
...
+
+                    monthlySummaries.Add(summary);
+                }
+            }
+
+            return monthlySummaries;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception("Error al obtener el resumen mensual", ex);
```
Comment (ignacio-villarroel-jalafund) [inline src/Services/UserService.cs]: Use the custom exections that we have in the project, or else the api will crash

File: src/Services/UserService.cs (hunk 4 above)
Comment (ignacio-villarroel-jalafund) [reply src/Services/UserService.cs]: Also change the exception to a custom one we have, if the existing execptions does not follow your requirements to throw an error create a new one that suits your error

File: src/Services/UserService.cs (hunk 3 above)
Comment (ignacio-villarroel-jalafund) [reply src/Services/UserService.cs]: Use a custom exception

File: src/Services/UserService.cs (hunk 2 above)
Comment (ignacio-villarroel-jalafund) [reply src/Services/UserService.cs]: The same way here, use a custom exception

File: src/Services/UserService.cs (hunk 1 above)
Comment (ronaldo-mendoza-jalafund) [reply src/Services/UserService.cs:48]: This will be applied later together with the FLUENT validation.

**Review by axel-ayala-jalafund — APPROVED** [review]

**Pull request merged**